structure_tests/structure_test.go

```diff
@@ -0,0 +1,100 @@
+package structure_tests
+
+import (
+	"io/ioutil"
+	"os/exec"
+	"strings"
+	"testing"
+	"encoding/json"
+	"flag"
+	"log"
+)
+
+type CommandTest struct {
+	Name				string		// name of test
+	Command				string 		// command to run
+	Flags				string 		// optional flags
+	ExpectedOutput		string 		// expected output of running command
+}
+
+type FileExistenceTest struct {
+	Name				string		// name of test
+	FilePath			string 		// file to check existence of
+	ShouldExist			bool 		// whether or not the file should exist
+}
+
+type FileContentTest struct {
+	Name				string		// name of test
+	FilePath			string 		// file to check existence of
+	ExpectedContents	string 		// expected contents of file	
+}
+
+type StructureTest struct {
+	Commands			[]CommandTest		`json:"commands"`
+	FileExistenceTests 	[]FileExistenceTest	`json:"file_existence"`
+	FileContentTests 	[]FileContentTest	`json:"file_contents"`
+}
+
+
+func TestRunCommand(t *testing.T) {
+	for _, tt := range tests.Commands {
+		t.Log(tt.Name)
+		var cmd *exec.Cmd
+		if tt.Flags != "" {
+			cmd = exec.Command(tt.Command, tt.Flags)
+		} else {
+			cmd = exec.Command(tt.Command)
+		}
+		actualOutput, err := cmd.CombinedOutput()
+		if err != nil {
+			t.Errorf("Error running command: %s %s. Error: %s. Output %s.", tt.Command, tt.Flags, err, actualOutput)
+		}
+		if tt.ExpectedOutput != "" && string(actualOutput) != tt.ExpectedOutput {
+			t.Errorf("Unexpected output for command: %s\nExpected: %s\nActual: %s", tt.Command, tt.ExpectedOutput, string(actualOutput))
+		}
+	}
+}
+
+
+func TestFileExists(t *testing.T) {
+	for _, tt := range tests.FileExistenceTests {
+		t.Log(tt.Name)
+		output, err := ioutil.ReadDir(tt.FilePath)
```

Comment:
What happens if this is a file, not a directory?

I think you might want to use: os.Stat: https://golang.org/pkg/os/#File.Stat